pkg/dns/azure/client/auth.go

```diff
@@ -38,78 +38,48 @@ func getAuthorizerForResource(config Config) (autorest.Authorizer, error) {
 		}
 	}
 
-	// Fallback to using tenant ID from env variable if not set.
-	if strings.TrimSpace(config.TenantID) == "" {
-		config.TenantID = os.Getenv("AZURE_TENANT_ID")
-		if strings.TrimSpace(config.TenantID) == "" {
-			return nil, errors.New("empty tenant ID")
+	// If the environment variable is set for any of the following, use the environment variable over any configuration
+	// value: AZURE_TENANT_ID, AZURE_CLIENT_ID, and AZURE_CLIENT_SECRET. The Azure SDK for Go function,
+	// NewDefaultAzureCredential, checks environment variables first.
+	if os.Getenv("AZURE_TENANT_ID") == "" {
+		err := os.Setenv("AZURE_TENANT_ID", config.TenantID)
+		if err != nil {
```

Comment:
The following is more idiomatic since you don't need the return value outside the `if` statement:
```go
		if err := os.Setenv("AZURE_TENANT_ID", config.TenantID); err != nil {
```
Same comment applies to other places where you add `err := ...; if err != nil`.